lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -2827,14 +2834,25 @@ func (d *zfs) MountVolumeSnapshot(snapVol Volume, op *operations.Operation) erro
 					d.logger.Debug("Regenerating filesystem UUID", logger.Ctx{"dev": volPath, "fs": tmpVolFsType})
 					err = regenerateFilesystemUUID(mountVol.ConfigBlockFilesystem(), volPath)
 					if err != nil {
-						return err
+						return nil, err
+					}
+				}
+			} else {
+				// ext4 will replay the journal if the filesystem is dirty.
+				// To prevent this kind of write access, we mount the ext4 filesystem
+				// with the ro,noload mount options.
+				// The noload option prevents the journal from being loaded on mounting.
+				if tmpVolFsType == "ext4" {
```

Comment:
I'd like to see this as a separate commit, along with explanation as to what its fixing. As I've taken a look at the LVM Ceph drivers and they don't have this behaviour.